### PR TITLE
feat: add binary decorator support

### PR DESCRIPTION
**Pull request by nicolasalexandre9**
### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

File: packages/forest_admin_datasource_customizer/lib/forest_admin_datasource_customizer/collection_customizer.rb [hunk 1]
```diff
@@ -250,6 +250,21 @@ def add_segment(name, definition)
...
+    # By default, all fields are transported as 'datauri', with the exception of primary and foreign
+    # keys.
+    #
+    # Using 'datauri' allows to use the FilePicker widget, while 'hex' is more suitable for
+    # short binary data (for instance binary uuids).
+    #
+    # @param name the name of the field
+    # @param binary_mode either 'datauri' or 'hex'
+    # @example
+    # .replaceFieldBinaryMode('avatar', 'datauri');
```
Comment (matthv) [inline packages/forest_admin_datasource_customizer/lib/forest_admin_datasource_customizer/collection_customizer.rb:263]: should be in snake case

File: packages/forest_admin_datasource_customizer/lib/forest_admin_datasource_customizer/decorators/rename_field/rename_field_collection_decorator.rb [hunk 2]
```diff
@@ -22,6 +22,15 @@ def rename_field(current_name, new_name)
 
           ForestAdminDatasourceToolkit::Validations::FieldValidator.validate_name(name, new_name)
 
+          # Revert previous renaming (avoids conflicts and need to recurse on this.toSubCollection).
```
Comment (matthv) [inline packages/forest_admin_datasource_customizer/lib/forest_admin_datasource_customizer/decorators/rename_field/rename_field_collection_decorator.rb:25]: same

**Review by matthv — APPROVED** [review]
Sounds good to me.
If you can just correct the method comments in camel case.

File: packages/forest_admin_datasource_customizer/lib/forest_admin_datasource_customizer/decorators/rename_field/rename_field_collection_decorator.rb (hunk 2 above)
Comment (nicolasalexandre9) [reply packages/forest_admin_datasource_customizer/lib/forest_admin_datasource_customizer/decorators/rename_field/rename_field_collection_decorator.rb]: done

**Review by matthv — APPROVED** [review]

**Pull request merged**